src/utils/utils.rs

```diff
@@ -156,6 +156,9 @@ pub(crate) fn download_file_with_resume(
                 // fault in case it was something odd which happened.
                 Some(DEK::HttpStatus(416)) => false,
                 Some(DEK::HttpStatus(400..=499)) | Some(DEK::FileNotFound) => true,
+                Some(DEK::IoError(io_err)) => {
+                    return Err(RustupError::IOError(std::io::Error::from(io_err.kind())).into());
```

Comment:
isn't io_err an IO error itself? Why do we need to construct a new one here?

Reply:
The `io_err` is an `&Error` and cannot be moved out, but I just realized we could return `Err(e)` in this match arm.

I'm also thinking about moving the logic here into `download::download_to_path_with_backend` and also moving these error variants into `DownloadError`, and not having to downcast `anyhow::Error` to a concrete error type. Would this be a good approach?

Reply:
If you want to experiment with it, and it turns out nicely, that does sound good.

Reply:
After investigation, I found some rustup logic relays on these error types. It would not be very beneficial to move these errors into download crate.